### PR TITLE
Pull content from Integration into the CMS

### PR DESCRIPTION
**Pull request by madz-lw**
Implements a 'Fetch now' button that pulls data from OpenAPI into the planning application fields in Sanity, if the Integrations option in Global Content is set to OpenAPI. Based on @duncanheron 's proof of concept. 
The integration option is set to manual by default, and displays the option to select Open API in the Global Content section.
Pulls in the following data: `application_type, name, development_address, development_description, location - latitude, longitude, applicationDocumentsUrl.` If the fields have a value it will display on Sanity.

Additionally shows success/error messages to user based on the status of the fetch. If the application number is invalid it reminds the user to check it.

TODO: Test needs to be re-enabled with mocks. There is an issue that keeps re-appearing: `SyntaxError: Cannot use import statement outside a module)`. It looks like an issue with how Jest is configured with ES module syntax. Will come back to it soon.

imageimage
image
image

### CODE REVIEW

File: sanity/schemas/planningApplication.ts [hunk 1]
```diff
@@ -36,6 +37,15 @@ export default defineType({
       validation: (Rule: any) => Rule.required(),
       readOnly: false,
     }),
+    defineField({
+      name: "populateApi",
+      title: "Auto populate from OpenData API",
+      type: "string",
+      // hidden: ({document}: any) => document?.datasource !== 'OpenData API',
```
Comment (duncanheron) [inline sanity/schemas/planningApplication.ts:44]: kill?

File: sanity/schemas/globalContent.ts [hunk 2]
```diff
@@ -5,6 +5,15 @@ export default defineType({
   name: "global-content",
   type: "document",
   fields: [
+    defineField({
+      title: "Integrations",
+      name: "integrations",
+      type: "string",
+      initialValue: "manual",
+      options: {
+        list: ["manual", "openAPI", "uniform"],
```
Comment (duncanheron) [inline sanity/schemas/globalContent.ts:14]: can we remove uniform until its a requirement

**Comment by duncanheron** [comment]
it would be slick if we could show a message when manual/no integration is chosen
"no integrations detected" or similar
image

File: sanity/schemas/planningApplication.ts (hunk 1 above)
Comment (madz-lw) [reply sanity/schemas/planningApplication.ts]: Removed ✨

File: sanity/schemas/globalContent.ts (hunk 2 above)
Comment (madz-lw) [reply sanity/schemas/globalContent.ts]: Removed ✨

**Comment by madz-lw** [comment]
> it would be slick if we could show a message when manual/no integration is chosen "no integrations detected" or similar image
> 
> Also the field label should be more generic "Auto populate from OpenData API" => "integrations" then the button can say "Fetch from {integration_type}"

**Review by duncanheron — APPROVED** [review]

**Pull request merged**